examples/crowdfunding/test/failing-tests.js

```diff
@@ -0,0 +1,437 @@
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        toAccountAddr: escrow.address,
+        amountMicroAlgos: goal - 1e6,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+  });
+
+  function updateAndOptIn () {
+    // update application with correct escrow account address
+    appArgs = [addressToPk(escrowAddress)]; // converts algorand address to Uint8Array
+
+    runtime.updateApp(
+      creator.address,
+      applicationId,
+      approvalProgram,
+      clearProgram,
+      {}, { appArgs: appArgs });
+
+    // opt-in to app
+    runtime.optInToApp(creator.address, applicationId, {}, {});
+    runtime.optInToApp(donor.address, applicationId, {}, {});
+  }
+
+  this.afterEach(async function () {
+    creator = new StoreAccount(initialCreatorBalance);
+    donor = new StoreAccount(initialDonorBalance);
+    runtime = new Runtime([master, creator, escrow, donor]);
+  });
+
+  it('should fail donation if donor has insufficient balance', () => {
+    updateAndOptIn();
+    appArgs = [stringToBytes('donate')];
+    const donationAmount = initialDonorBalance + 1000;
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        toAccountAddr: escrow.address,
+        amountMicroAlgos: donationAmount,
+        payFlags: { totalFee: 1000 }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), 'RUNTIME_ERR1401: Cannot withdraw');
+  });
+
+  it('should fail donation if timestamp is after endDate', () => {
+    updateAndOptIn();
+    // set timestamp to after of endDate
+    runtime.setRoundAndTimestamp(5, endDate.getSeconds() + 100);
+
+    assert.throws(() => runtime.executeTx(donateTxGroup), rejectMsg);
+  });
+
+  it('should fail donation if timestamp is before of beginDate', () => {
+    updateAndOptIn();
+    // set timestamp to out of endDate
+    runtime.setRoundAndTimestamp(5, beginDate.getSeconds() - 100);
+
+    assert.throws(() => runtime.executeTx(donateTxGroup), rejectMsg);
+  });
+
+  it('should fail if goal is met, and donor tries to reclaim funds', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    runtime.setRoundAndTimestamp(5, endDate.getTime() + 100); // end date is passed
+
+    // donor should be able to reclaim if goal is NOT met
```

Comment:
Please update this comment, this is a failing scenario

Reply:
yes